src/Eval.hs

```diff
@@ -397,11 +398,11 @@ eval ctx xobj@(XObj o info ty) preference resolver =
     evaluateApp :: Evaluator
     evaluateApp (AppPat f' args) =
       case f' of
-        (ListPat l _) -> go l
-        (SymPat sym _) -> go sym
+        (ListPat l _) -> go l ResolveLocal
+        (SymPat sym _) -> go sym resolver
         _ -> pure (evalError ctx (format (GenericMalformed xobj)) (xobjInfo xobj))
-      where go x =
-              do (newCtx, f) <- eval ctx x preference ResolveLocal
+      where go x resolve =
+              do (newCtx, f) <- eval ctx x preference resolve
```

Comment:
Are these changes part of the fix? They seem unrelated at a first glance

Reply:
Yes, they are. I didn't think it'd be relevant either, but if we use `ResolveLocal` on resolution of the `Sym` it results in hangs.